TodoListClient/Startup.cs

```diff
@@ -119,6 +119,10 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
             app.UseEndpoints(endpoints =>
             {
+                endpoints.MapControllerRoute(
+                    name: "IndexRoute",
+                    pattern: "{controller}/{action}/{id?}");
```

Comment:
This was the problem.

.NET 6.0 treats 'Index' as a blank route and doesn't create the URL to load the proper controller actions by default. What happened was it would fall through to the `default` route and load the home page instead.